client/src/components/feedback/send.js

```diff
@@ -0,0 +1,260 @@
+    this.sendApi = this.sendApi.bind(this);
+    this.cancel = this.cancel.bind(this);
+    this.showFeedbackSent = this.showFeedbackSent.bind(this);
+    this.showFeedbackNotSent = this.showFeedbackNotSent.bind(this);
+    this.showTimer = this.showTimer.bind(this);
+  }
+
+  onSubmit() {
+    if (this.state.validInput) {
+      if (this.state.description.length < 10) {
+        this.tooShortFeedback();
+      } else if (this.state.description.length > 400) {
+        this.tooLongFeedback();
+      } else {
+        this.sendApi();
+      }
+    }
+  }
+
+  setInputFalse() {
+    this.setState(() => {
+      this.setState({
+        validInput: false,
+      });
+    });
+  }
+
+  validateDescription() {
+    const { description } = this.state;
+    const regex = /(^[a-zA-Z1-9]+[a-zA-Z1-9., ]+)$/;
+    const match = regex.test(description);
+    if (description.length > 1 && match !== false) {
+      // eslint-disable-next-line no-unused-expressions
+      description > 400
+        ? this.setState({
+          tooLongFeedback: true,
+          validInput: true,
+        })
+        : this.setState({
+          tooLongFeedback: false,
+          attemptCount: false,
+          validInput: true,
+        });
+      // eslint-disable-next-line no-mixed-operators
+    } else if (description.length === 0
+      // eslint-disable-next-line no-mixed-operators
+      || description.length === 1 && /([a-zA-Z1-9])/.test(description)) {
+      this.setState({
+        validInput: true,
+      });
+    } else {
+      this.setInputFalse();
+    }
+  }
+
+  handleChange(e) {
+    this.setState({ description: e.target.value, tooShortFeedback: false }, () => {
+      this.validateDescription();
+    });
+  }
+
+  tooShortFeedback() {
+    const currentCallId = Math.random();
+    this.setState({
+      currentCallId,
+      tooShortFeedback: true,
+    });
+    setTimeout(() => {
+      if (currentCallId !== this.state.currentCallId) return;
+      this.setState({
+        tooShortFeedback: false,
+      });
+    }, 3000);
+  }
+
+  tooLongFeedback() {
+    const currentCallId = Math.random();
+    this.setState({
+      currentCallId,
+      attemptCount: true,
+      tooLongFeedback: true,
+    });
+    setTimeout(() => {
+      if (currentCallId !== this.state.currentCallId) return;
+      this.setState({
+        tooLongFeedback: true,
+        attemptCount: false,
+      });
+    }, 120);
+  }
+
+  sendApi() {
+    const currentCallId = Math.random();
+    this.setState({
+      isLoading: true,
+      tooShortFeedback: false,
+      tooLongFeedback: false,
+      showTimer: true,
+      inputDisabled: true,
+      currentCallId,
+    });
+    setTimeout(() => {
+      if (currentCallId !== this.state.currentCallId) return;
+      this.setState({
+        showTimer: false,
+      });
+    }, 4000);
+    setTimeout(() => {
+      if (currentCallId !== this.state.currentCallId) return;
+      Api.post('feedback', {
+        personId: 555,
```

Comment:
As I understand, Feedback can only send a registered user? I changed the personId to those that I have in the database, then the feedback is sent.

Reply:
Yes. This hardcoded solution is just till we finish our authorization.